src/Type.h

```diff
@@ -270,13 +270,64 @@ class Type : public Obj {
 	void SetName(const std::string& arg_name) { name = arg_name; }
 	const std::string& GetName() const { return name; }
 
-	typedef std::map<std::string, std::set<Type*> > TypeAliasMap;
+	struct TypePtrComparer {
+		bool operator()(const TypePtr& a, const TypePtr& b) const
+			{ return a.get() < b.get(); }
+	};
+	using TypePtrSet = std::set<TypePtr, TypePtrComparer>;
+	using TypeAliasMap = std::map<std::string, TypePtrSet, std::less<>>;
+
+	/**
+	 * Returns a mapping of type-name to all other type names declared as
+	 * an alias to it.
+	 */
+	static const TypeAliasMap& GetAliasMap()
+		{ return type_aliases; }
+
+	/**
+	 * Returns true if the given type name has any declared aliases
+	 */
+	static bool HasAliases(std::string_view type_name)
+		{ return Type::type_aliases.find(type_name) != Type::type_aliases.end(); }
 
+	/**
+	 * Returns the set of all type names declared as an aliases to the given
+	 * type name.  A static empty set is returned if there are no aliases.
+	 */
+	static const TypePtrSet& Aliases(std::string_view type_name)
+		{
+		static TypePtrSet empty;
+		auto it = Type::type_aliases.find(type_name);
+		return it == Type::type_aliases.end() ? empty : it->second;
+		}
+
+	[[deprecated("Remove in v4.1. Use zeek::Type::Aliases() instead.")]]
 	static std::set<Type*> GetAliases(const std::string& type_name)
-		{ return Type::type_aliases[type_name]; }
+		{
+		std::set<Type*> rval;
+		for ( const auto& t : Type::type_aliases[type_name] )
+			rval.emplace(t.get());
+		return rval;
+		}
+
+	/**
+	 * Registers a new type alias.
+	 * @param type_name  the name of the type to register a new alias for.
+	 * @param type  the associated alias type of *type_name*.
+	 * @return  true if the alias is now registered or false if the alias was
+	 * already previously registered.
+	 */
+	static bool RegisterAlias(std::string_view type_name, TypePtr type)
```

Comment:
is it necessary to change & deprecate the name (instead of just overloading)?

Reply:
Deprecate: not necessary, but probably preferable to have the one taking `Type*` go away.

Rename: not necessary, but mainly because the return type to `bool` making overload not possible.  Change in return type is not strictly necessary, but seems natural.  Also, "add" doesn't reflect as well as "register" the expected/actual behavior if called twice w/ same type name (uniqueness implications).  The "add" version is actually a "replace" anyway (misleading name, and also not a purpose I ever wanted it to be used for).

Reply:
Actually, scratch most of that naming explain, think I got mixed up with something else and conjured it up retrospectively.  Only reason must have been to change return type.